### PR TITLE
Agency Transaction Builder Tests

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose

This PR adds tests for the Agency Transaction Builder, which was previously untested.

- [x] :hammer: Refactoring/simplification

#### Backports:

- [x] No backports required

### Testing & Verification

This PR adds tests.

### CODE REVIEW

**Comment by maierlars** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/13866/

File: arangod/Agency/TransactionBuilder.h [hunk 1]
```diff
@@ -110,9 +116,9 @@ struct envelope {
       return std::move(*this);
     }
 
-    envelope end() {
+    envelope end(std::string clientId = {}) {
```
Comment (jsteemann) [inline arangod/Agency/TransactionBuilder.h:119]:
```suggestion
    envelope end(std::string const& clientId = {}) {
```

File: arangod/Agency/TransactionBuilder.h [hunk 2]
```diff
@@ -134,10 +140,10 @@ struct envelope {
   };
 
   struct write_trx {
-    envelope end() {
+    envelope end(std::string clientId = {}) {
```
Comment (jsteemann) [inline arangod/Agency/TransactionBuilder.h:143]:
```suggestion
    envelope end(std::string const& clientId = {}) {
```

**Review by jsteemann — APPROVED** [review]

**Comment by maierlars** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/13867/

**Pull request merged**